### PR TITLE
fix: In certain conditions, don't add a booking to busyTimes

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

Seated events should not block themselves until fully booked.

### CODE REVIEW

**Review by alannnc — DISMISSED** [review]
Works and very well documented!

**Review by zomars — APPROVED** [review]

**Review by zomars — CHANGES_REQUESTED** [review]
Blocking as the e2e failure is a legit one

**Review by zomars — APPROVED** [review]

File: packages/core/getBusyTimes.ts [hunk 1]
```diff
@@ -74,42 +74,57 @@ export async function getBusyTimes(params: {
...
+      seatsReferences: true,
+    },
+  });
+
+  const busyTimes = bookings.reduce(
+    (aggregate: EventBusyDetails[], { id, startTime, endTime, eventType, title, seatsReferences }) => {
+      // Seat references on the current event are non-blocking until the event is fully booked.
+      if (
+        // this is purely for jest; prisma responses are []
+        seatsReferences &&
```
Comment (alannnc) [inline packages/core/getBusyTimes.ts:118]:
```suggestion
        seatsReferences &&
        // and seatReferences is greater than 0
        seatsReferences.length > 0 &&
```

Comment (alannnc) [reply packages/core/getBusyTimes.ts:118]: @emrysal found this while on my branch.

**Comment by emrysal** [comment]
Work continued as part of https://github.com/calcom/cal.com/pull/9778

**Pull request closed without merging**